aix/AixProcessList.c

```diff
@@ -0,0 +1,296 @@
+/*
+htop - AixProcessList.c
+(C) 2014 Hisham H. Muhammad
+(C) 2018 Calvin Buckley
+Released under the GNU GPL, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "ProcessList.h"
+#include "AixProcess.h"
+#include "AixProcessList.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <string.h>
+#include <time.h>
+#include <math.h>
+#include <sys/types.h>
+#include <sys/proc.h>
+#include <procinfo.h>
+#ifndef __PASE__
+#include <libperfstat.h>
+#else
+#include <sys/vminfo.h>
+#endif
+
+/*{
+
+#ifndef __PASE__
+#include <libperfstat.h>
+#endif
+
+// publically consumed
+typedef struct CPUData_ {
+   // per libperfstat.h, clock ticks
+   unsigned long long utime, stime, itime, wtime;
+   // warning: doubles are 4 bytes in structs on AIX
+   // ...not like we need precision here anyways
+   double utime_p, stime_p, itime_p, wtime_p;
+} CPUData;
+
+typedef struct AixProcessList_ {
+   ProcessList super;
+   CPUData* cpus;
+#ifndef __PASE__
+   perfstat_cpu_t* ps_cpus;
+   perfstat_cpu_total_t ps_ct;
+#endif
+} AixProcessList;
+
+#ifndef Process_isKernelThread
+#define Process_isKernelThread(_process) (_process->kernel == 1)
+#endif
+
+#ifndef Process_isUserlandThread
+// XXX
+#define Process_isUserlandThread(_process) (false)
+#endif
+
+}*/
```

Comment:
More commented code from the header file that could be deleted.

Reply:
This is actually how htop's buid system works - the headers have things from the comments transcluded into them, from what I understand (!).

Reply:
Ah, interesting!